src/wcl/rope.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
Please write a comment describing what rope is/does

Reply:
Yeah, I need to not be a hypocrite about this. We need more comments in the code even if we're writing higher quality code comments are important.

Reply:
Done. I put them directly above the `rope` and `rope_builder` classes so that they show up in vscode